st2common/st2common/models/db/__init__.py

```diff
@@ -163,7 +163,10 @@ def update(self, instance, **kwargs):
         return instance.update(**kwargs)
 
     def delete(self, instance):
-        instance.delete()
+        return instance.delete()
+
+    def delete_by_query(self, **query):
+        return self.model.objects.filter(**query).delete()
```

Comment:
You can modify this method a bit so we also log the query and profile data:

```python
result = self.model.objects.filter(**query).delete()
log_query_and_profile_data_for_queryset(result)
result = result.delete()
return result
```

It's slightly annoying that you can't simply pass result of `delete` to this method, but mongoengine API is not ideal and not all the methods return an instance of `QuerySet` object. That's also one of the reasons why right now we only log "find()" queries.